src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -11,41 +12,136 @@
 
     private ReadFilterLibrary(){ /*no instance*/ }
 
-    public static final ReadFilter ALLOW_ALL_READS = read -> true;
-
-    public static final ReadFilter MAPPED =  read -> ! read.isUnmapped();
-    public static final ReadFilter PRIMARY_ALIGNMENT = read -> ! read.isSecondaryAlignment();
-    public static final ReadFilter NOT_DUPLICATE = read -> ! read.isDuplicate();
-    public static final ReadFilter PASSES_VENDOR_QUALITY_CHECK = read -> ! read.failsVendorQualityCheck();
-    public static final ReadFilter MAPPING_QUALITY_AVAILABLE = read -> read.getMappingQuality() != QualityUtils.MAPPING_QUALITY_UNAVAILABLE;
-    public static final ReadFilter MAPPING_QUALITY_NOT_ZERO = read -> read.getMappingQuality() != 0;
+    /**
+     * local classes for static read filters
+     */
+    public static class AllowAllReads extends ReadFilter {
+        private static final long serialVersionUID = 1L;
+        @Override public boolean test(final GATKRead read){return true;}}
 
     //Note: do not call getCigar to avoid creation of new Cigar objects
-    public static final ReadFilter READLENGTH_EQUALS_CIGARLENGTH = read -> read.isUnmapped() || read.getLength() == Cigar.getReadLength(read.getCigarElements());
+    public static class CigarIsSupported extends ReadFilter {
```

Comment:
Should these all have ReadFilter as part of their name?  It's a bit inconsistent on the commandline that some do and some don't.